internal/clients/substrate/substrate.go

```diff
@@ -0,0 +1,74 @@
+package substrate
+
+import (
+	"fmt"
+	"math/big"
+	"strconv"
+
+	"github.com/threefoldtech/tf-kyc-verifier/internal/logger"
+
+	// use tfchain go client
+
+	tfchain "github.com/threefoldtech/tfchain/clients/tfchain-client-go"
+)
+
+type Substrate struct {
+	api    *tfchain.Substrate
+	logger logger.Logger
+}
+
+func New(config SubstrateConfig, logger logger.Logger) (*Substrate, error) {
+	mgr := tfchain.NewManager(config.GetWsProviderURL())
+	api, err := mgr.Substrate()
+	if err != nil {
+		return nil, fmt.Errorf("substrate connection error: failed to initialize Substrate client: %w", err)
```

Comment:
please review the error messages to comply with https://preslav.me/2023/04/14/golang-error-handling-is-a-form-of-storytelling/